From 63a8262bd4dfdc7a54559aa61069eec5e6d7087d Mon Sep 17 00:00:00 2001
From: Chuanhua Han <chuanhua.han@nxp.com>
Date: Thu, 13 Dec 2018 12:10:50 +0800
Subject: [PATCH 402/706] arm64: dts: ls1028a: add dspi controller DT nodes

Add the dspi support on ls1028a.

Signed-off-by: Chuanhua Han <chuanhua.han@nxp.com>
(cherry picked from commit 9fd983e182753f8a187990721d27be43b2077d8a)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 25 ++++++++++++++-----
 1 file changed, 19 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index 3eb4e2b048ae..af8db8fe5339 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -335,8 +335,8 @@
 			clocks = <&sysclk>;
 		};
 
-		dspi0: dspi@2100000 {
-			compatible = "fsl,ls1021a-v1.0-dspi";
+		dspi0: spi@2100000 {
+			compatible = "fsl,ls1028a-dspi", "fsl,ls1021a-v1.0-dspi";
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0x0 0x2100000 0x0 0x10000>;
@@ -344,12 +344,12 @@
 			clock-names = "dspi";
 			clocks = <&clockgen 4 1>;
 			spi-num-chipselects = <5>;
-			big-endian;
+			little-endian;
 			status = "disabled";
 		};
 
-		dspi1: dspi@2110000 {
-			compatible = "fsl,ls1021a-v1.0-dspi";
+		dspi1: spi@2110000 {
+			compatible = "fsl,ls1028a-dspi", "fsl,ls1021a-v1.0-dspi";
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0x0 0x2110000 0x0 0x10000>;
@@ -357,7 +357,20 @@
 			clock-names = "dspi";
 			clocks = <&clockgen 4 1>;
 			spi-num-chipselects = <5>;
-			big-endian;
+			little-endian;
+			status = "disabled";
+		};
+
+		dspi2: spi@2120000 {
+			compatible = "fsl,ls1028a-dspi", "fsl,ls1021a-v1.0-dspi";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0x0 0x2120000 0x0 0x10000>;
+			interrupts = <GIC_SPI 26 IRQ_TYPE_LEVEL_HIGH>;
+			clock-names = "dspi";
+			clocks = <&clockgen 4 1>;
+			spi-num-chipselects = <5>;
+			little-endian;
 			status = "disabled";
 		};
 
-- 
2.17.1

